File: readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>


int main(void)
{
    // promt
    string text = get_string("Text: ");

    // letters
    int Letters = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        char c = text[i];
        if (isalpha(c) != 0)
        {
            Letters++;
        }
    }


    // words
    int Words = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == ' ')
        {
            Words++;
        }
    }



    // sentences
    int Sentences = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            Sentences ++;
        }
    }


    // calculations
    float L = ((float)Letters / (float)Words) * 100;
    float S = ((float)Sentences / (float)Words) * 100;
    float subindex = 0.0588 * L - 0.296 * S - 15.8;
    int index = round(subindex);
    if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }

}

File: readability.py
# TODO
from cs50 import get_string

letters = 0
words = 0
sentences = 0

text = get_string("Text: ")

for i in range(len(text)):
    if(text[i].isalpha()):
        letters += 1

    if (text[i].isspace()):
        words +=1

    if (text[i] == "." or text[i] == "!" or text[i] == "?"):
        sentences += 1

level = 0.0588 * (letters / words * 100) - 0.296 * (sentences / words * 100) - 15.8

print(f"Grade {round(level)}")
